FreeRTOS-Plus/Demo/FreeRTOS_Plus_PKCS11_Windows_Simulator/examples/objects.c

```diff
@@ -0,0 +1,405 @@
+                                          prime1, PRIME_1_LENGTH + 1,
+                                          prime2, PRIME_2_LENGTH + 1,
+                                          d, D_LENGTH + 1,
+                                          e, E_LENGTH + 1 );
+    configASSERT( lMbedResult == 0 );
+
+    /* Export Exponent 1, Exponent 2, Coefficient. */
+    lMbedResult = mbedtls_mpi_write_binary( ( mbedtls_mpi const * ) &pxRsaContext->DP,
+                                            exponent1,
+                                            EXPONENT_1_LENGTH + 1 );
+    configASSERT( lMbedResult == 0 );
+
+    lMbedResult = mbedtls_mpi_write_binary( ( mbedtls_mpi const * ) &pxRsaContext->DQ,
+                                            exponent2,
+                                            EXPONENT_2_LENGTH + 1 );
+    configASSERT( lMbedResult == 0 );
+
+    lMbedResult = mbedtls_mpi_write_binary( ( mbedtls_mpi const * ) &pxRsaContext->QP,
+                                            coefficient,
+                                            COEFFICIENT_LENGTH + 1 );
+    configASSERT( lMbedResult == 0 );
+
+    /*
+     * Now we have created a template of CK_ATTRIBUTEs that describe the structure
+     * of the RSA private key we want to create. We will pass this to the Cryptoki
+     * library, as well as it's length, to create the described private key on
+     * the token.
+     *
+     * The pointers to the various RSA parameters are incremented by one, in
+     * order to remove the 0 padding if it was added, since we use the original
+     * length of the RSA parameter.
+     *
+     */
+    CK_ATTRIBUTE xPrivateKeyTemplate[] =
+    {
+        { CKA_CLASS,            &xPrivateKeyClass, sizeof( CK_OBJECT_CLASS ) },
+        { CKA_KEY_TYPE,         &xPrivateKeyType,  sizeof( CK_KEY_TYPE )     },
+        { CKA_LABEL,            pxLabel,           sizeof( pxLabel )         },
+        { CKA_TOKEN,            &xTrue,            sizeof( CK_BBOOL )        },
+        { CKA_SIGN,             &xTrue,            sizeof( CK_BBOOL )        },
+        { CKA_MODULUS,          modulus + 1,       MODULUS_LENGTH            },
+        { CKA_PRIVATE_EXPONENT, d + 1,             D_LENGTH                  },
+        { CKA_PUBLIC_EXPONENT,  e + 1,             E_LENGTH                  },
+        { CKA_PRIME_1,          prime1 + 1,        PRIME_1_LENGTH            },
+        { CKA_PRIME_2,          prime2 + 1,        PRIME_2_LENGTH            },
+        { CKA_EXPONENT_1,       exponent1 + 1,     EXPONENT_1_LENGTH         },
+        { CKA_EXPONENT_2,       exponent2 + 1,     EXPONENT_2_LENGTH         },
+        { CKA_COEFFICIENT,      coefficient + 1,   COEFFICIENT_LENGTH        }
+    };
+
+    configPRINTF( ( "Creating private key with label: %s \r\n",
+                    pkcs11configLABEL_DEVICE_PRIVATE_KEY_FOR_TLS ) );
+    configPRINTF( ( "FreeRTOS_P11_Key.dat has been created in the Visual Studio" \
```

Comment:
Have we created ```FreeRTOS_P11_Key.dat``` until this point?

Reply:
Oops, I'll move this printf to after the call to create.